app/controllers/file_controller.rb

```diff
@@ -4,8 +4,8 @@ class FileController < ApplicationController
   before_action :authenticate_request!
 
   def create_submit
-    return head :unauthorized unless @current_user
     return head :forbidden unless User.can_submit_to?(@current_user['id'], params[:id])
+    return head :BadRequest if ContestProblem.hard_overdue?(params[:id])
```

Comment:
`BadRequest` a nie `Forbidden` z powodu przekierowywania we froncie? Zastanawiam się, czy 403 w naszym przypadku też powinno wylogowywać użytkownika. W których przypadkach back-end wyrzuca 403? Robiłem wyszukiwanie i znalazło omawiamy fragment i ten: https://github.com/zmora-agh/zmora-server-rails/blob/a8ed426728983d8beee3854d72d119cdb38b826b/app/middlewares/authorization_middleware.rb#L16 ale ten wyjątek nie jest nigdzie jawnie tworzony?